code/modules/mapping/reader.dm

```diff
@@ -265,127 +508,201 @@
-					ycrd--
 					MAPLOADING_CHECK_TICK
+					++xcrd
 					continue
-
-				var/list/cache = modelCache[line]
+				var/list/cache = modelCache[model_key]
 				if(!cache)
 					SSatoms.map_loader_stop()
-					CRASH("Undefined model key in DMM: [line]")
-				build_coordinate(cache, locate(true_xcrd, ycrd, zcrd), no_afterchange, placeOnTop)
+					CRASH("Undefined model key in DMM: [model_key]")
+				build_coordinate(cache, locate(xcrd, ycrd, zcrd), no_afterchange, placeOnTop)
 
 				// only bother with bounds that actually exist
 				if(!first_found)
 					first_found = TRUE
+					first_x = xcrd
 					first_y = ycrd
+				last_x = xcrd
 				last_y = ycrd
-				ycrd--
-				MAPLOADING_CHECK_TICK
-			// The x coord never changes, so this is safe
-			if(first_found)
-				first_x = true_xcrd
-				last_x = true_xcrd
-		else
-			// This is the dmm parser, note the double loop
-			for(var/i in 1 + y_starting_skip to line_count - y_ending_skip)
-				var/line = gset.gridLines[i]
-
-				var/xcrd = true_xcrd
-				for(var/tpos in 1 + x_starting_skip to x_target step key_len)
-					var/model_key = copytext(line, tpos, tpos + key_len)
-					if(model_key == space_key && no_afterchange)
-						#ifdef TESTING
-						++turfsSkipped
-						#endif
-						MAPLOADING_CHECK_TICK
-						++xcrd
-						continue
-					var/list/cache = modelCache[model_key]
-					if(!cache)
-						SSatoms.map_loader_stop()
-						CRASH("Undefined model key in DMM: [model_key]")
-					build_coordinate(cache, locate(xcrd, ycrd, zcrd), no_afterchange, placeOnTop)
-
-					// only bother with bounds that actually exist
-					if(!first_found)
-						first_found = TRUE
-						first_x = xcrd
-						first_y = ycrd
-					last_x = xcrd
-					last_y = ycrd
-					MAPLOADING_CHECK_TICK
-					++xcrd
-				ycrd--
 				MAPLOADING_CHECK_TICK
+				++xcrd
+			ycrd--
+			MAPLOADING_CHECK_TICK
 		bounds[MAP_MINX] = min(bounds[MAP_MINX], first_x)
 		bounds[MAP_MINY] = min(bounds[MAP_MINY], last_y)
 		bounds[MAP_MINZ] = min(bounds[MAP_MINZ], zcrd)
 		bounds[MAP_MAXX] = max(bounds[MAP_MAXX], last_x)
 		bounds[MAP_MAXY] = max(bounds[MAP_MAXY], first_y)
 		bounds[MAP_MAXZ] = max(bounds[MAP_MAXZ], zcrd)
 
-	// And we are done lads, call it off
-	SSatoms.map_loader_stop()
-	if(!no_changeturf)
-		for(var/turf/T as anything in block(locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]), locate(bounds[MAP_MAXX], bounds[MAP_MAXY], bounds[MAP_MAXZ])))
-			//we do this after we load everything in. if we don't; we'll have weird atmos bugs regarding atmos adjacent turfs
-			T.AfterChange(CHANGETURF_IGNORE_AIR)
+	return TRUE
 
-	if(has_expanded_world_maxx || has_expanded_world_maxy)
-		SEND_GLOBAL_SIGNAL(COMSIG_GLOB_EXPANDED_WORLD_BOUNDS, has_expanded_world_maxx, has_expanded_world_maxy)
+GLOBAL_LIST_EMPTY(map_model_default)
 
-	#ifdef TESTING
-	if(turfsSkipped)
-		testing("Skipped loading [turfsSkipped] default turfs")
-	#endif
+/datum/parsed_map/proc/build_cache(no_changeturf, bad_paths)
+	if(map_format == MAP_TGM)
```

Comment:
Same here, you should be doing an exhaustive switch

Reply:
see above